frontend/lib/rh/rental-history.tsx

```diff
@@ -198,74 +199,139 @@ function RentalHistoryForm(): JSX.Element {
         )}
       </SessionUpdatingFormSubmitter>
       <ClearSessionButton
-        to={JustfixRoutes.locale.rh.splash}
+        to={props.prevStep}
         portalRef={cancelControlRef}
         label={li18n._(t`Cancel request`)}
       />
       <Route
         path={JustfixRoutes.locale.rh.formAddressModal}
         exact
-        render={() => (
-          <FormConfirmAddressModal toStep2={JustfixRoutes.locale.rh.preview} />
-        )}
+        render={() => <FormConfirmAddressModal toStep2={props.nextStep} />}
       />
     </Page>
   );
-}
+});
 
-function RentalHistoryPreview(): JSX.Element {
+const RentalHistoryRsUnitCheck = MiddleProgressStep((props) => {
+  const rsInfo = useContext(AppContext).session.rentStabInfo;
   return (
     <Page
-      title={li18n._(t`Review your request to the DHCR`)}
+      title={
+        rsInfo && rsInfo.latestYear && rsInfo.latestUnitCount
+          ? li18n._(t`It looks like your apartment may be rent stabilized!`)
+          : li18n._(t`It’s unlikely that your apartment is rent stabilized`)
+      }
       withHeading
       className="content"
     >
-      <p>
-        <Trans>
-          Here is a preview of the request for your Rent History. It includes
-          your address and apartment number so that the DHCR can mail you.
-        </Trans>
-      </p>
-      <ForeignLanguageOnly>
-        <p className="is-uppercase is-size-7">
-          <InYourLanguageTranslation />{" "}
-          <Trans>(Note: the request will be sent in English)</Trans>
-        </p>
-      </ForeignLanguageOnly>
-      <article className="message">
-        <div className="message-header has-text-weight-normal">
-          <Trans>
-            To: New York Division of Housing and Community Renewal (DHCR)
-          </Trans>
-        </div>
-        <div className="message-body content">
-          <RhEmailToDhcr />
-        </div>
-      </article>
-      <DemoDeploymentNote>
-        <p>
-          This demo site <strong>will not send</strong> a real request to the
-          DHCR.
-        </p>
-      </DemoDeploymentNote>
+      {rsInfo && rsInfo.latestYear && rsInfo.latestUnitCount ? (
+        <>
+          <p>
+            <Trans>
+              Your building had{" "}
+              <Plural
+                value={rsInfo.latestUnitCount}
+                one="1 rent stabilized unit"
+                other="# rent stabilized units"
+              />{" "}
```

Comment:
@toolness for some reason this plural tag isn't getting recognized by lingui when running the extract command, and I can't for the life of me figure out why. Any help would be life changing!

Reply:
I think this is because for non-trivial internationalization (e.g. not just simple string substitution), you actually need to run _both_ `yarn lingui:extract` _and_ `yarn lingui:compile`.  This is because at an implementation level, non-trivial i18n is actually accomplished via a JS function that `yarn lingui:compile` generates and puts into the locale bundle file.  Without that pre-compiled functionality, there can be no pluralization or other fancy non-trivial i18n.

Reply:
That said, ideally our build pipeline should be running `yarn lingui:compile` whenever it notices that the `.po` files have changed!  That way developers wouldn't have to remember to do this step in such non-trivial situations... I've filed #1780 to address this later.

Reply:
@toolness so I've tried compiling and I think the problem is deeper than that? See below— for some reason the plural tag isn't getting written into our messages.po files at all. lmk if I'm missing something